src/services/stringCompletions.ts

```diff
@@ -1106,11 +1189,29 @@ function getModulesForPathsPattern(
     const directories = normalizedSuffix
         ? emptyArray
         : mapDefined(tryGetDirectories(host, baseDirectory), dir => dir === "node_modules" ? undefined : directoryResult(dir));
-    return [...matches, ...directories];
 
-    function trimPrefixAndSuffix(path: string): string | undefined {
+    if (inputBaseDirectory && inputBaseDirectory !== baseDirectory) {

```

Comment:
TODO: a helper has to be introduced to deduplicate this branch with the ones that creates original `matches`